### DISCUSSION TITLE
Dialog child component in SimpleComponent using tracker?

### DISCUSSION THREAD

**Discussion by gtamas** [Relm4/Relm4#554] [General]
I have a widget which is supposed to open dialog when the user clicks to a button.
This widget (the one that opens the dialog) is a SimpleComponent

Normally I would do something like this:

```
pub struct SomeModel {
    ...
    dialog: Controller<SaveDialogModel>
}
```

This works, but now I have:

```
#[tracker::track]
#[derive(Debug, PartialEq)]
pub struct SomeModelUiModel {
    ...
    save: Controller<SaveDialogModel>
}
```

However this doesn't work because of `#[tracker::track]` and `PartialEq`. I'm getting an error:

`binary operation `==` cannot be applied to type `Controller<SaveDialogModel>`

If I remove track and derive annotations, then it's fine but I need the tracker here. How to solve this?

**Comment by AaronErhardt** [comment]
Tracker has a special feature for your use-case. You need the `#[no_eq]` attribute to make your code work: https://docs.rs/tracker/latest/tracker/#further-attributes

**Reply by gtamas** [reply]
Thanks for the info, this solved it!
